pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -285,6 +288,25 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 
 	pLog.Debug("install job still running")
 
+	if time.Since(job.CreationTimestamp.Time) > podStatusCheckDelay {
+		installPod, err := r.getInstallPod(job, pLog)
+		if err != nil {
+			pLog.WithError(err).Error("could not get install pod")
+			return r.transitionStage(instance, hivev1.ClusterProvisionStageFailed, "InstallPodMissing", err.Error(), pLog)
```

Comment:
We cannot transition if the pod cannot be found. We may reconcile the ClusterProvision after the Job has been created but before the Pod has been created.